src/services/refactors/convertFunctionToEs6Class.ts

```diff
@@ -0,0 +1,193 @@
+        const start = typeof positionOrRange === "number" ? positionOrRange : positionOrRange.pos;
+        const sourceFile = context.file;
+        const checker = context.program.getTypeChecker();
+        const token = getTokenAtPosition(sourceFile, start);
+        const ctorSymbol = checker.getSymbolAtLocation(token);
+
+        const deletes: (() => any)[] = [];
+
+        if (!(ctorSymbol.flags & (SymbolFlags.Function | SymbolFlags.Variable))) {
+            return [];
+        }
+
+        const ctorDeclaration = ctorSymbol.valueDeclaration;
+        const changeTracker = textChanges.ChangeTracker.fromCodeFixContext(context);
+
+        let precedingNode: Node;
+        let newClassDeclaration: ClassDeclaration;
+        switch (ctorDeclaration.kind) {
+            case SyntaxKind.FunctionDeclaration:
+                precedingNode = ctorDeclaration;
+                deletes.push(() => changeTracker.deleteNode(sourceFile, ctorDeclaration));
+                newClassDeclaration = createClassFromFunctionDeclaration(ctorDeclaration as FunctionDeclaration);
+                break;
+
+            case SyntaxKind.VariableDeclaration:
+                precedingNode = ctorDeclaration.parent.parent;
+                if ((<VariableDeclarationList>ctorDeclaration.parent).declarations.length === 1) {
+                    deletes.push(() => changeTracker.deleteNode(sourceFile, precedingNode));
+                }
+                else {
+                    deletes.push(() => changeTracker.deleteNodeInList(sourceFile, ctorDeclaration));
+                }
+                newClassDeclaration = createClassFromVariableDeclaration(ctorDeclaration as VariableDeclaration);
+                break;
+        }
+
+        if (!newClassDeclaration) {
+            return [];
+        }
+
+        // Because the preceding node could be touched, we need to insert nodes before delete nodes.
+        changeTracker.insertNodeAfter(sourceFile, precedingNode, newClassDeclaration, { suffix: "\n" });
+        for (const deleteCallback of deletes) {
+            deleteCallback();
+        }
+
+        return [{
+            description: `Convert function ${ctorSymbol.name} to ES6 class`,
+            changes: changeTracker.getChanges()
+        }];
+
+        function createClassElementsFromSymbol(symbol: Symbol) {
+            const memberElements: ClassElement[] = [];
+            // all instance members are stored in the "member" array of symbol
+            if (symbol.members) {
+                symbol.members.forEach(member => {
+                    const memberElement = createClassElement(member, /*modifiers*/ undefined);
+                    if (memberElement) {
+                        memberElements.push(memberElement);
+                    }
+                });
+            }
+
+            // all static members are stored in the "exports" array of symbol
+            if (symbol.exports) {

```

Comment:
also for member initialization, I wounder if we can even do these correctellly... 

```ts
var c = function() { this.list.push(1); }
c.prototype.list = [];
``` 
is not the same as:

```ts
class c {
    constructor() { this.list.push(1); }
    list = [];
```

let a side that it is not valid JS strictly speaking.